### PR TITLE
SPIKE AUTH - rush part 2

### PR DESCRIPTION
**Pull request by carolinapapes**
This pull request introduces significant improvements to the authentication API documentation, validation, and smoke tests, along with some minor configuration and import cleanups. The main focus is on clarifying and extending the OpenAPI documentation for authentication endpoints, adding new endpoints for email verification and password reset, and ensuring that the smoke tests cover these new behaviors and edge cases.

**Authentication API documentation and endpoint improvements:**

* Expanded and clarified OpenAPI documentation for all `/auth` endpoints, including more descriptive summaries, response descriptions, and error handling for signup, login, password recovery, resend verification, and OAuth flows. New endpoints `/auth/verify-email` and `/auth/reset-password` are documented with appropriate request and response schemas. [[1]](diffhunk://#diff-a1497cfd010e94829aa2ea6f8aa6f0139eca2a39c43b2c6e57eb0587e09298e2L38-R38) [[2]](diffhunk://#diff-a1497cfd010e94829aa2ea6f8aa6f0139eca2a39c43b2c6e57eb0587e09298e2L48-R48) [[3]](diffhunk://#diff-a1497cfd010e94829aa2ea6f8aa6f0139eca2a39c43b2c6e57eb0587e09298e2L75-R75) [[4]](diffhunk://#diff-a1497cfd010e94829aa2ea6f8aa6f0139eca2a39c43b2c6e57eb0587e09298e2L85-R103) [[5]](diffhunk://#diff-a1497cfd010e94829aa2ea6f8aa6f0139eca2a39c43b2c6e57eb0587e09298e2L124-R124) [[6]](diffhunk://#diff-a1497cfd010e94829aa2ea6f8aa6f0139eca2a39c43b2c6e57eb0587e09298e2L134-R134) [[7]](diffhunk://#diff-a1497cfd010e94829aa2ea6f8aa6f0139eca2a39c43b2c6e57eb0587e09298e2L155-R157) [[8]](diffhunk://#diff-a1497cfd010e94829aa2ea6f8aa6f0139eca2a39c43b2c6e57eb0587e09298e2L165-R240) [[9]](diffhunk://#diff-a1497cfd010e94829aa2ea6f8aa6f0139eca2a39c43b2c6e57eb0587e09298e2L198-R295) [[10]](diffhunk://#diff-a1497cfd010e94829aa2ea6f8aa6f0139eca2a39c43b2c6e57eb0587e09298e2L222-R308) [[11]](diffhunk://#diff-a1497cfd010e94829aa2ea6f8aa6f0139eca2a39c43b2c6e57eb0587e09298e2R1110-R1136)

**Smoke tests for authentication edge cases:**

* Added new smoke tests for signup validation errors, signup rate limiting, invalid email verification and password reset tokens, and the public, non-enumerating resend verification endpoint. These tests ensure that the backend correctly handles invalid or malicious requests and maintains user privacy. [[1]](diffhunk://#diff-e5da61c59effadc0e7ff23a180d9fcf2d76b7da2d0ae106ddb59a028816da870R75-R132) [[2]](diffhunk://#diff-e5da61c59effadc0e7ff23a180d9fcf2d76b7da2d0ae106ddb59a028816da870L208-R266) [[3]](diffhunk://#diff-e5da61c59effadc0e7ff23a180d9fcf2d76b7da2d0ae106ddb59a028816da870R311-R387) [[4]](diffhunk://#diff-e5da61c59effadc0e7ff23a180d9fcf2d76b7da2d0ae106ddb59a028816da870R414-R424)

**OpenAPI schema enhancements:**

* Added new request schemas for `VerifyEmailRequest` and `ResetPasswordRequest`, including field validation constraints. Updated existing schemas to explicitly disallow additional properties for stricter validation.

**Configuration and import cleanups:**

* Updated `.devcontainer/devcontainer.json` to focus on the Next.js frontend by default, with improved VSCode settings and extensions. The backend and sockets configurations are now commented out for clarity. [[1]](diffhunk://#diff-24ad71c8613ddcf6fd23818cb3bb477a1fb6d83af4550b0bad43099813088686L1-L46) [[2]](diffhunk://#diff-24ad71c8613ddcf6fd23818cb3bb477a1fb6d83af4550b0bad43099813088686R20-R69)
* Fixed imports in backend scripts and app initialization to reference the correct authentication service modules. [[1]](diffhunk://#diff-13e05d2f8181d748cba393c73fdbaae5e021dba16c2d21a83bdfabc9c5d6dfd3L2-R2) [[2]](diffhunk://#diff-fb433806c7b7e665f41f8e78ee376580b3f8faf3472eb4115655109679f973faL10-R10)

### CODE REVIEW

File: containers/backend/app/src/auth/local/local.routes.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+export const localRouter = Router();
+
+localRouter.post('/signup', validateBody(SignupReqSchema), localSignupEmailRateLimit, postSignup);
+localRouter.post(
+  '/login',
+  localLoginIpRateLimit,
+  validateBody(LoginReqSchema),
+  localLoginIdentifierRateLimit,
+  postLogin,
```
[the remark this answers is not in the document]
Comment (carolinapapes) [reply containers/backend/app/src/auth/local/local.routes.ts:22]: localLoginIpRateLimit its literally here

File: containers/backend/app/src/auth/verification/verification.routes.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+export const verificationRouter = Router();
+
+verificationRouter.post(
+  '/resend-verification',
+  validateBody(ResendVerificationReqSchema),
+  verificationResendEmailRateLimit,
+  postResendVerification,
+);
+verificationRouter.post('/verify-email', validateBody(VerifyEmailReqSchema), postVerifyEmail);
```
[the remark this answers is not in the document]
Comment (carolinapapes) [reply containers/backend/app/src/auth/verification/verification.routes.ts:17]: verificationResendEmailRateLimit,?

File: containers/frontend/web/src/components/composites/api-feedback/api-feedback.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+  if (result.ok || result?.api?.ok)
+    return (
+      <Text variant="body-xs" className={apiFeedbackStyles.success}>
+        {successMessage}
+      </Text>
+    );
+
+  const errorKey = result?.api?.error?.code || (result?.error.code as Parameters<typeof tError>[0]);
+
```
[the remark this answers is not in the document]
Comment (carolinapapes) [reply containers/frontend/web/src/components/composites/api-feedback/api-feedback.tsx:41]: Will fix on refactor of this component after finish auth.

File: containers/frontend/web/src/features/auth/logout/logout.action.ts [hunk 4]
```diff
@@ -4,9 +4,13 @@ import type { LogoutRes } from '@/contracts/api/auth/auth.contract';
...
+  try {
+    const response = await fetchClient<LogoutRes>('/api/auth/logout', 'POST', undefined, {
+      withAuth: true,
+    });
 
-  return response;
+    return response.data;
+  } catch {
+    return { ok: false, error: { code: 'AUTH_FAILED_TO_LOGOUT' } };
+  }
```
[the remark this answers is not in the document]
Comment (carolinapapes) [reply containers/frontend/web/src/features/auth/logout/logout.action.ts:15]: fixed

File: containers/frontend/web/src/i18n/messages/es.json [hunk 5]
```diff
@@ -15,6 +15,9 @@
     "textAreaField": {
       "characterCount": "{current} de {max} caracteres"
     },
+    "submitButton": {
+      "pendingLabel": "CHANGE_LATER"
+    },
```
[the remark this answers is not in the document]
Comment (carolinapapes) [reply containers/frontend/web/src/i18n/messages/es.json:20]: fixed

File: containers/frontend/web/src/features/auth/resend-verification/resend-verification.action.ts [hunk 6]
```diff
@@ -1,22 +1,31 @@
...
-  try {
-    const cookie = (await cookies()).toString();
-    const { data } = await fetchServer<ApiResponse<unknown>>(
+  const result = await withServerAction(async () => {
+    const cookieStore = await cookies();
+    const cookie = cookieStore.toString();
+    const email = cookieStore.get(PENDING_VERIFICATION_EMAIL_COOKIE)?.value;
+    if (!email) {
+      return { ok: false, error: { code: 'AUTH_NO_PENDING_VERIFICATION' } } as const;
+    }
```
[the remark this answers is not in the document]
Comment (carolinapapes) [reply containers/frontend/web/src/features/auth/resend-verification/resend-verification.action.ts]: fixed

File: containers/frontend/web/src/features/auth/verify-email/verify-email.client.tsx [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+
+  const data = (await res.json()) as VerifyEmailRes;
+
+  if (data.ok) return { kind: 'success' };
+
+  if (data.error.code === 'VALIDATION_ERROR') {
+    return { kind: 'validation-error', key: data.error.details?.fields.token?.[0] ?? 'REQUIRED' };
+  }
+
+  return { kind: 'api-error', code: data.error.code };
```
[the remark this answers is not in the document]
Comment (carolinapapes) [reply containers/frontend/web/src/features/auth/verify-email/verify-email.client.tsx]: fixed

File: containers/frontend/web/src/features/auth/recover/recover.success.tsx [hunk 8]
```diff
@@ -0,0 +1,17 @@
+import { getTranslations } from 'next-intl/server';
+
+import { InternalLink, Stack } from '@components';
+import { MessageBlock } from '@components/composites/text-block/text-block';
+import { RecoverResend } from './recover.resend';
+
+export default async function RecoverSuccess() {
+  const t = await getTranslations('features.auth.recover');
+
```
[the remark this answers is not in the document]
Comment (carolinapapes) [reply containers/frontend/web/src/features/auth/recover/recover.success.tsx:9]: fixed

File: containers/frontend/web/src/i18n/messages/ca.json [hunk 9]
```diff
@@ -12,6 +12,9 @@
       "es": "ES",
       "ca": "CA"
     },
+    "submitButton": {
+      "pendingLabel": "CHANGE_LATER"
+    },
```
[the remark this answers is not in the document]
Comment (carolinapapes) [reply containers/frontend/web/src/i18n/messages/ca.json:17]: fixed

File: containers/frontend/web/src/app/[locale]/(guest)/(auth)/recover/success/page.tsx [hunk 10]
```diff
@@ -0,0 +1,5 @@
+import RecoverSuccess from '@/features/auth/recover/recover.sucess';
```
[the remark this answers is not in the document]
Comment (carolinapapes) [reply containers/frontend/web/src/app/[locale]/(guest)/(auth)/recover/success/page.tsx]: fixed

File: containers/frontend/web/src/app/auth/verify-email/page.tsx [hunk 11]
```diff
@@ -0,0 +1,5 @@
+import { VerifyEmailFeature } from '@/features/auth/verify-email';
+
+export default function VerifyEmailPage() {
+  return <VerifyEmailFeature />;
+}
```
[the remark this answers is not in the document]
Comment (carolinapapes) [reply containers/frontend/web/src/app/auth/verify-email/page.tsx:5]: will fix but they are needed for now

File: containers/frontend/web/src/app/auth/reset-password/page.tsx [hunk 12]
```diff
@@ -0,0 +1,5 @@
+import { ResetPasswordFeature } from '@/features/auth/reset-password';
+
+export default function ResetPasswordPage() {
+  return <ResetPasswordFeature />;
+}
```
[the remark this answers is not in the document]
Comment (carolinapapes) [reply containers/frontend/web/src/app/auth/reset-password/page.tsx:5]: will fix but they are needed for now

File: containers/contracts/api/auth/auth.contract.ts [hunk 13]
```diff
@@ -82,6 +79,59 @@ export type ResendVerificationError = (typeof AUTH_RESEND_VERIFICATION_ERRORS)[n
...
+  'VALIDATION_ERROR',
+] as const satisfies readonly AuthErrorName[];
+
+export type ResetPasswordError = (typeof AUTH_RESET_PASSWORD_ERRORS)[number];
+
+export type ResetPasswordRes = ApiResponse<
+  ResetPasswordOk,
+  ResetPasswordError,
+  ValidationErrorDetails
+>;
```
[the remark this answers is not in the document]
Comment (carolinapapes) [reply containers/contracts/api/auth/auth.contract.ts:118]: fixed

File: containers/backend/app/docs/openapi.yaml [hunk 14]
```diff
@@ -82,25 +82,25 @@ paths:
...
         '403':
-          description: Account locked or email not verified
+          description: Account locked
           content:
             application/json:
               schema:
                 $ref: '#/components/schemas/AuthErrorResponse'
         '429':
-          description: Rate limited
+          description: Rate limited (too many failed attempts)
```
[the remark this answers is not in the document]
Comment (carolinapapes) [reply containers/backend/app/docs/openapi.yaml:109]: fixed

[the remark this answers is not in the document]
Comment (carolinapapes) [reply containers/backend/app/docs/openapi.yaml:109]: fixed

File: containers/backend/app/docs/openapi.yaml [hunk 15]
```diff
@@ -162,19 +163,81 @@ paths:
...
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ValidationErrorResponse'
+        '500':
+          description: Internal error
           content:
             application/json:
               schema:
                 $ref: '#/components/schemas/AuthErrorResponse'
```
[the remark this answers is not in the document]
Comment (carolinapapes) [reply containers/backend/app/docs/openapi.yaml:200]: fixed

File: containers/backend/app/docs/openapi.yaml [hunk 16]
```diff
@@ -162,19 +163,81 @@ paths:
...
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ValidationErrorResponse'
+        '500':
+          description: Internal error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/AuthErrorResponse'
```
[the remark this answers is not in the document]
Comment (carolinapapes) [reply containers/backend/app/docs/openapi.yaml:243]: fixed

File: containers/backend/app/src/auth/local/local.controller.ts [hunk 17]
```diff
@@ -0,0 +1,69 @@
...
+  req.session.destroy((err) => {
+    if (err) return sendError(res, 'AUTH_INTERNAL_ERROR');
+
+    res.clearCookie('sid', {
+      path: '/',
+      sameSite: 'lax',
+    });
+
+    sendOk(res, { user: null }, 200);
+  });
```
[the remark this answers is not in the document]
Comment (carolinapapes) [reply containers/backend/app/src/auth/local/local.controller.ts:68]: fixed

File: containers/frontend/web/src/components/composites/api-feedback/api-feedback.tsx [hunk 18]
```diff
@@ -0,0 +1,47 @@
...
+}: ApiFeedbackProps<TCode>) {
+  if (!result) return null;
+
+  const tError = useTranslations('errors');
+
+  if (result.ok || result?.api?.ok)
+    return (
+      <Text variant="body-xs" className={apiFeedbackStyles.success}>
+        {successMessage}
+      </Text>
```
[the remark this answers is not in the document]
Comment (carolinapapes) [reply containers/frontend/web/src/components/composites/api-feedback/api-feedback.tsx:37]: Will fix later

File: containers/frontend/web/src/components/composites/api-feedback/api-feedback.tsx [hunk 19]
```diff
@@ -0,0 +1,47 @@
...
+    );
+
+  const errorKey = result?.api?.error?.code || (result?.error.code as Parameters<typeof tError>[0]);
+
+  return (
+    <Text variant="body-xs" className={apiFeedbackStyles.error}>
+      {tError(errorKey)}
+    </Text>
```
[the remark this answers is not in the document]
Comment (carolinapapes) [reply containers/frontend/web/src/components/composites/api-feedback/api-feedback.tsx:45]: Will fix later

File: containers/frontend/web/src/app/[locale]/(guest)/(auth)/recover/success/page.tsx (hunk 10 above)
[the remark this answers is not in the document]
Comment (carolinapapes) [reply containers/frontend/web/src/app/[locale]/(guest)/(auth)/recover/success/page.tsx]: Will dix later

File: containers/backend/app/docs/openapi.yaml [hunk 20]
```diff
@@ -131,7 +131,7 @@ paths:
               $ref: '#/components/schemas/RecoverRequest'
       responses:
         '200':
-          description: Recovery email flow accepted
+          description: Request accepted; password reset email sent if user exists
           content:
             application/json:
               schema:
```
[the remark this answers is not in the document]
Comment (carolinapapes) [reply containers/backend/app/docs/openapi.yaml:143]: Will fix later

File: containers/backend/app/docs/openapi.yaml [hunk 21]
```diff
@@ -162,19 +163,93 @@ paths:
               $ref: '#/components/schemas/ResendVerificationRequest'
       responses:
         '200':
-          description: Verification email sent
+          description: Request accepted (verification email sent or hidden for privacy)
           content:
             application/json:
               schema:
```
[the remark this answers is not in the document]
Comment (carolinapapes) [reply containers/backend/app/docs/openapi.yaml:181]: Will fix later

File: containers/backend/app/docs/openapi.yaml [hunk 22]
```diff
@@ -45,7 +45,7 @@ paths:
               $ref: '#/components/schemas/SignupRequest'
       responses:
         '200':
-          description: Account created
+          description: Account created; verification token sent to email
           content:
             application/json:
               schema:
```
[the remark this answers is not in the document]
Comment (carolinapapes) [reply containers/backend/app/docs/openapi.yaml:51]: Will fix later

File: containers/frontend/web/src/app/[locale]/(guest)/(auth)/recover/success/page.tsx (hunk 10 above)
[the remark this answers is not in the document]
Comment (carolinapapes) [reply containers/frontend/web/src/app/[locale]/(guest)/(auth)/recover/success/page.tsx]: fixed

File: containers/backend/app/docs/openapi.yaml (hunk 20 above)
[the remark this answers is not in the document]
Comment (carolinapapes) [reply containers/backend/app/docs/openapi.yaml:143]: fixed

File: containers/backend/app/docs/openapi.yaml (hunk 22 above)
[the remark this answers is not in the document]
Comment (carolinapapes) [reply containers/backend/app/docs/openapi.yaml:51]: fixed

File: containers/backend/app/docs/openapi.yaml (hunk 21 above)
[the remark this answers is not in the document]
Comment (carolinapapes) [reply containers/backend/app/docs/openapi.yaml:181]: fixed.
